File: src/FoodStuffs.Web/ClientApp/src/models/api/recipes/SaveRecipeRequest.js
export default class SaveRecipeRequest {
  constructor() {
    this.id = 0;
    this.name = '';
    this.ingredients = '';
    this.directions = '';
    this.cookTimeMinutes = 0;
    this.prepTimeMinutes = 0;
    this.categories = [];
  }
}

File: src/FoodStuffs.Model/Events/Recipes/SaveRecipeRequest.cs
namespace FoodStuffs.Model.Events.Recipes;

public record SaveRecipeRequest(
    int Id,
    string Name,
    string Directions,
    string Sides,
    int? CookTimeMinutes,
    int? PrepTimeMinutes,
    bool IsForMealPlanning,
    List<SaveRecipeRequestIngredient> Ingredients,
    List<SaveRecipeRequestShoppingItem> ShoppingItems,
    List<string> Categories);
